CONTRIBUTING.md

```diff
@@ -0,0 +1,226 @@
+
+# Check ESLint rules
+npm run lint
+
+# Auto-fix formatting
+npm run format
+```
+
+## Making Changes
+
+1. **Create a feature branch**:
+
+   ```bash
+   git checkout -b feature/your-feature-name
+   ```
+
+2. **Make your changes**:
+   - Write clean, readable code
+   - Follow existing patterns in the codebase
+   - Add tests for new features
+   - Update documentation as needed
+
+3. **Commit your changes**:
+
+   ```bash
+   git add .
+   git commit -m "Brief description of your changes"
+   ```
+
+   Husky will automatically run lint-staged, which will:
+   - Format your code with Prettier
+   - Run ESLint with auto-fix
+   - Only process files you've staged for commit
+
+4. **Keep your branch updated**:
+   ```bash
+   git fetch origin
+   git rebase origin/main
+   ```
+
+## Testing
+
+We use Web Test Runner for testing our Lit components.
+
+### Running Tests
+
+```bash
+# Run all tests with coverage
+npm test
+
+# Run tests in watch mode (great for TDD)
+npm run test:watch
+
+# Run tests for CI
+npm run test:ci
+```
+
+### Writing Tests
+
+- Place test files next to the code they test with `.test.js` extension
+- Use descriptive test names that explain what is being tested
+- Test user interactions, not implementation details
+- Aim for high coverage of critical paths
+
+Example test structure:
+
+```javascript
+import { fixture, html, expect } from '@open-wc/testing';
+import './your-component.js';
+
+describe('YourComponent', () => {
+  it('renders with default properties', async () => {
+    const el = await fixture(html`<your-component></your-component>`);
+    expect(el.shadowRoot.querySelector('.container')).to.exist;
+  });
+
+  it('handles user interaction', async () => {
+    const el = await fixture(html`<your-component></your-component>`);
+    const button = el.shadowRoot.querySelector('button');
+    button.click();
+    await el.updateComplete;
+    // Assert expected behavior
+  });
+});
+```
+
+## Submitting Changes
+
+1. **Push your branch**:
+
+   ```bash
+   git push origin feature/your-feature-name
+   ```
+
+2. **Create a Pull Request**:
+   - Go to the repository on GitHub
+   - Click "New Pull Request"
+   - Select your feature branch
+   - Fill out the PR template with:
+     - Clear description of changes
+     - Link to related issues
+     - Screenshots (if UI changes)
+     - Testing performed
+
+3. **Wait for CI checks**:
+   - GitHub Actions will automatically run tests and checks
+   - All checks must pass before merging
+   - Fix any issues reported by CI
+
+## Code Review Process
+
+1. **Automated Checks**:
+   - Prettier formatting validation
+   - Test suite execution
+   - Build verification
+   - Bundle size check
+
+2. **Human Review**:
+   - At least one maintainer will review your PR
+   - Respond to feedback constructively
+   - Make requested changes in new commits
+   - Mark conversations as resolved once addressed
+
+3. **Merging**:
+   - PRs are squash-merged to keep history clean
```

Comment:
do we squash merge? I think the default is merge commit, which keeps the history

Reply:
I've switched the default to squash-merge.